src/graph.py

```diff
@@ -348,14 +348,18 @@ def output_forest_as_networkx(self, vertex_indices, edge_indices):
         # Replace the edge indices with the actual edges (source name, target name) by indexing into the interactome
         edges = self.interactome_dataframe.loc[edge_indices]
         forest = nx.from_pandas_edgelist(edges, 'source', 'target', edge_attr=True)
-        # the above won't capture the singletons, so we'll add them here
+        # the above won't capture the singletons, so we'll add them here (JL: Do we want singletons though?)
         forest.add_nodes_from(list(set(self.nodes[vertex_indices]) - set(forest.nodes())))
 
         # Set node degrees as attributes on nodes in the netowrkx graph
         nx.set_node_attributes(forest, pd.DataFrame(self.node_degrees, index=self.nodes, columns=['degree']).loc[list(forest.nodes())].astype(int).to_dict(orient='index'))
 
-        # Set all othe attributes on graph
-        nx.set_node_attributes(forest, self.node_attributes.loc[list(forest.nodes())].dropna(how='all').to_dict(orient='index'))
+        # Set all the attributes on graph
+        # If there is no overlap between the two (i.e. no prizes are included in forest, 
+        # which can happen via randomizations), then the next line throws an error. So this is to check for that.
+        # This is clunky though - one solution is to reindex self.node_attributes 
+        if len(set(self.node_attributes.index) & set(forest.nodes())) > 0:
+            nx.set_node_attributes(forest, self.node_attributes.loc[list(forest.nodes())].dropna(how='all').to_dict(orient='index'))
```

Comment:
Please change this to the `.reindex`

Reply:
e966c3b

Reply:
e966c3b520e404e28a1aafce5425b7f407bdbbbe

Reply:
e966c3b520e404e28a1aafce5425b7f407bdbbbe